gammapy/scripts/image_pipe.py

```diff
@@ -68,12 +82,43 @@ def make_counts(self, obs_id):
         counts_map.data = counts_map.data.value
         self.maps["counts"] = counts_map
 
-    def make_total_counts(self):
+    def _low_counts(self, obs_id, ncounts_min=0):
+        """Check if there are events for this observation after applying the offset and energy range
+
+        Parameters
+        ----------
+        obs_id: int
+            Observation ID
+        ncounts_min : int
+            Minimum counts required for one observation
+        Returns
+        -------
+        bool :
+            False if there is no events for this observation for the energy range used to compute the image
+
+        """
+        obs = self.data_store.obs(obs_id=obs_id)
+        events = obs.events.select_energy(self.energy_band)
+        events = events.select_offset(self.offset_band)
+        if len(events) <= ncounts_min:
+            log.warning('Skipping obs ' + str(obs_id) + ' because it only has ' + str(ncounts_min) + ' counts.')
+            return False
+        else:
+            return True
+
+    def make_total_counts(self, ncounts_min):
```

Comment:
Here you forgot to put a default for `ncounts_min`.
Generally it's annying for the user to have to pass ncounts_min many times, and it's hard to keep the different defaults consistent.

How about making it a parameter that is set once in the constructor and stored as a data member, that is accessed from `_low_counts`?